File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxnlnrn.py
"""Reference emit for IDXNLNRN — INDEXED START NOT LESS + READ NEXT → 78.00."""
# EXPECTED: 78.00
ROWS = {10: 12.50, 20: 8.00, 42: 25.00, 55: 30.00, 99: 15.00}
START = 20
# GnuCOBOL BDB: START NOT LESS positions at first key >= START; NEXT includes it.
total = sum(v for k, v in ROWS.items() if int(k) >= START)
print(f"{total:.2f}")

File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxnlnrn.java
// EXPECTED: 78.00
// Reference emit for IDXNLNRN — INDEXED START NOT LESS + READ NEXT.
import java.util.*;
public final class IdxnlnrnRef {
  public static void main(String[] args) {
    Map<Integer, Double> rows = new HashMap<>();
    rows.put(10, 12.50); rows.put(20, 8.00); rows.put(42, 25.00);
    rows.put(55, 30.00); rows.put(99, 15.00);
    int start = 20;
    double total = 0;
    for (Map.Entry<Integer, Double> e : rows.entrySet()) {
      if (e.getKey() >= start) total += e.getValue();
    }
    System.out.printf(Locale.US, "%.2f%n", total);
  }
}
